test/UnitTests/Utils.cs

```diff
@@ -44,5 +45,32 @@ public static string BytesToHexString(byte[] bytes)
             }
             return hexString;  
         }
+
+        // Reflection is used to test inaccessible methods
+        public static object RunInstanceMethod(Type type, string strMethod, object objInstance, object[] objParams)
```

Comment:
Don't love this, but agree that there doesn't appear to be a way to mock the `GetObjectResponse` and inject it into the right place otherwise.

Reply:
Me neither :/ I couldn't find a better way to do it